File: hw/bonilla/U3/HW21TemplateMethod/TemplateMethod/src/main/java/ec/edu/espe/templatemethod/model/Tea.java
package ec.edu.espe.templatemethod.model;

import java.util.Scanner;

/**
 *
 * @author Bonilla David SoftCrafters
 */
public class Tea extends CaffeineBeverage {

    public void brew() {
        System.out.println("Steep the tea");
    }

    public void addCondiments() {
        System.out.println("Adding lemon");
    }

    public boolean wantsCondiments() { // a hook operation 
        String answer = getUserInput();
        if (answer.toLowerCase().startsWith("y")) {
            return true;
        } else {
            return false;
        }
    }

    private String getUserInput() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Would you like lemon with your tea (y/n)? ");
        String answer = scanner.nextLine();
        return answer;
    }
}

File: hw/fuentes/U3/HW21TEMPLATE/HW21TEMPLATEPY/ec/edu/espe/cafe/model/Tea.py
from model.CaffeineBeverage import CaffeineBeverage

class Tea(CaffeineBeverage):
    def brew(self):
        print("Steeping the tea")

    def addCondiments(self):
        print("Adding lemon")

    def wantsCondiments(self):
        answer = self.getUserInput()
        return answer is not None and answer.lower().startswith("y")

    def getUserInput(self):
        try:
            answer = input("Would you like lemon with your tea (y/n)? ")
            return answer
        except Exception:
            print("Error reading input.")
            return None
